### ISSUE TITLE
SGD keeps blowing up

### ISSUE THREAD

**Issue by qthen** [kuroware/candle#2]
SGD is unable to optimize the baseline neural network (a simple shallow two hidden layer fully connected layer with `ReLU`). The training loss more or less stays the same and changing the learning rate seems to have no effect after trying values from `0.1` to `1e-6`. 

It seems the SGD with momentum is also no better, strangely, adaptive learning optimizers like Adam don't have this problem. Furthermore, this problem only starts to happen when the number of PCA components is above 30, which seems strange since the error signal should be stronger.

**Comment by qthen** [comment]
It's a learning rate problem (at least shallowly). After some more playing around (and many iterations), `1e-6` on SGD with momentum does begin lowering the loss to reasonable values that is comparable to `Adam`.
